### ISSUE TITLE
failure searching for CHECKSUM

### ISSUE THREAD

**Issue by cookie33** [EUDAT-B2SAFE/B2HANDLE#69]
Hi,

When searching for a URL it works.
```
./epicclient2.py os /home/robertv/test/credentials_surfsara_epic3_certificates_new_841  search URL www.test1.com
["841/cf537936-0d42-11e6-ad5f-5254000df0ed"]
```

The handle looks as follows:
```
./epicclient2.py os /home/robertv/test/credentials_surfsara_epic3_certificates_new_841  read 841/cf537936-0d42-11e6-ad5f-5254000df0ed
[{"index": 1, "ttl": 86400, "type": "URL", "timestamp": "2016-04-29T05:47:46Z", "data": {"value": "www.test1.com", "format": "string"}}, {"index": 2, "ttl": 86400, "type": "TEST", "timestamp": "2016-04-29T05:48:20Z", "data": {"value": "this is a test", "format": "string"}}, {"index": 3, "ttl": 86400, "type": "CHECKSUM", "timestamp": "2016-04-29T07:39:17Z", "data": {"value": "12345678A", "format": "string"}}, {"index": 100, "ttl": 86400, "type": "HS_ADMIN", "timestamp": "2016-04-28T13:12:08Z", "data": {"value": {"index": 200, "handle": "0.NA/841", "permissions": "011111110011"}, "format": "admin"}}]
```
When searching for a CHECKSUM it fails
```
./epicclient2.py os /home/robertv/test/credentials_surfsara_epic3_certificates_new_841  search CHECKSUM 12345678A
Traceback (most recent call last):
  File "./epicclient2.py", line 297, in <module>
    _ARGS.func(_ARGS)
  File "./epicclient2.py", line 56, in search
    result = client.search_handle(**kvpairs)
  File "/usr/lib/python2.6/site-packages/b2handle/handleclient.py", line 957, in search_handle
    list_of_handles = self.__searcher.search_handle(URL=URL, prefix=prefix, **key_value_pairs)
  File "/usr/lib/python2.6/site-packages/b2handle/searcher.py", line 208, in search_handle
    query = self.create_revlookup_query(*fulltext_searchterms, **args)
  File "/usr/lib/python2.6/site-packages/b2handle/searcher.py", line 320, in create_revlookup_query
    query = query+'&'+key+'='+value
TypeError: cannot concatenate 'str' and 'NoneType' objects
```
What is wrong?

Greetings,

Robert Verkerk

**Comment by merretbuurman** [comment]
Hi,
the case of searching only for checksum without any URL had not been foreseen. I've fixed this. It will be merged as soon as integration tests have passed.

**Comment by cookie33** [comment]
Basic checks performed with the new fix:
```
./epicclient2.py os /home/robertv/test/credentials_surfsara_epic3_certificates_new_841 read 841/631e9f3a-1066-11e6-8b8c-5254000df0ed
[{"index": 1, "ttl": 86400, "type": "URL", "timestamp": "2016-05-02T13:04:21Z", "data": {"value": "www.test.com", "format": "string"}}, {"index": 2, "ttl": 86400, "type": "CHECKSUM", "timestamp": "2016-05-02T13:04:21Z", "data": {"value": "123456789", "format": "string"}}, {"index": 100, "ttl": 86400, "type": "HS_ADMIN", "timestamp": "2016-05-02T13:04:21Z", "data": {"value": {"index": 200, "handle": "0.NA/841", "permissions": "011111110011"}, "format": "admin"}}]
```
and searching for CHECKSUM now works:
```
./epicclient2.py os /home/robertv/test/credentials_surfsara_epic3_certificates_new_841  search CHECKSUM "123456789"
["841/631e9f3a-1066-11e6-8b8c-5254000df0ed"]
```
and some further tests:
```
./epicclient2.py os /home/robertv/test/credentials_surfsara_epic3_certificates_new_841 modify 841/631e9f3a-1066-11e6-8b8c-5254000df0ed CHECKSUM This_is_a_test
True

./epicclient2.py os /home/robertv/test/credentials_surfsara_epic3_certificates_new_841  search CHECKSUM "This*"
["841/631e9f3a-1066-11e6-8b8c-5254000df0ed"]

./epicclient2.py os /home/robertv/test/credentials_surfsara_epic3_certificates_new_841 read 841/631e9f3a-1066-11e6-8b8c-5254000df0ed
[{"index": 1, "ttl": 86400, "type": "URL", "timestamp": "2016-05-02T13:04:21Z", "data": {"value": "www.test.com", "format": "string"}}, {"index": 2, "ttl": 86400, "type": "CHECKSUM", "timestamp": "2016-05-02T13:21:24Z", "data": {"value": "This_is_a_test", "format": "string"}}, {"index": 100, "ttl": 86400, "type": "HS_ADMIN", "timestamp": "2016-05-02T13:04:21Z", "data": {"value": {"index": 200, "handle": "0.NA/841", "permissions": "011111110011"}, "format": "admin"}}]
```
and searching for an URL:
```
./epicclient2.py os /home/robertv/test/credentials_surfsara_epic3_certificates_new_841  search URL "www.test.com"
["841/00048aac-f7db-11e5-829b-5254000df0ed", "841/002aa872-f7db-11e5-8824-5254000df0ed", "841/00510760-f7db-11e5-ba3b-5254000df0ed", "841/00772d64-f7db-11e5-87d1-5254000df0ed", "841/009cbd5e-f7db-11e5-a255-5254000df0ed", "841/00d6d886-f7db-11e5-b253-5254000df0ed", "841/00fd07e0-f7db-11e5-92fc-5254000df0ed", "841/01184526-9cd4-11e5-8fea-040091643bd0new", "841/01184526-9cd4-11e5-8fea-040091643bd0newnew", "841/0122c0ca-f7db-11e5-b931-5254000df0ed", "841/0149bb26-f7db-11e5-8a54-5254000df0ed", "841/017102a8-f7db-11e5-befa-5254000df0ed", "841/01976880-f7db-11e5-bd06-5254000df0ed", "841/01bd967c-f7db-11e5-a66e-5254000df0ed", "841/01e24846-f7db-11e5-b0b2-5254000df0ed", "841/02083812-f7db-11e5-95a7-5254000df0ed", "841/0246cdca-f7db-11e5-80d0-5254000df0ed", "841/02832d74-f7db-11e5-b606-5254000df0ed", "841/02abd1c0-f7db-11e5-aeb6-5254000df0ed", "841/02d1e22a-f7db-11e5-bbf3-5254000df0ed",.... "841/ff4592b4-f7da-11e5-b301-5254000df0ed", "841/ff6ae5aa-f7da-11e5-b9fc-5254000df0ed", "841/ffa4130c-f7da-11e5-ba66-5254000df0ed", "841/ffdfba10-f7da-11e5-b1f3-5254000df0ed", "843/2e8f473a-ffc0-11e5-b9c3-5254000df0ed"]
```

**Comment by merretbuurman** [comment]
Thanks for testing this!

**Comment by merretbuurman** [comment]
Fixed in PR #72 . Will close this when PR #72 is merged.

**Comment by merretbuurman** [comment]
Closing, as PR #72 is merged.
